tests/tests/check_session.py

```diff
@@ -12,16 +12,40 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from ducktape.tests.session import generate_results_dir
+from ducktape.tests.session import generate_results_dir, SessionContext
+from ducktape.cluster.localhost import LocalhostCluster
 
 import os.path
+import pickle
+import shutil
 import tempfile
 
 
 class CheckGenerateResultsDir(object):
+    def setup_method(self, _):
+        self.tempdir = tempfile.mkdtemp()
+
     def check_generate_results_root(self):
         """Check the generated results directory has the specified path as its root"""
         results_root = os.path.abspath(tempfile.mkdtemp())
         results_dir = generate_results_dir(results_root, "my_session_id")
         assert results_dir.find(results_root) == 0
 
+    def check_pickleable(self):
+        """Check that session_context object is pickleable
+        This is necessary so that SessionContext objects can be shared between processes when using python
+        multiprocessing module.
+        """
+
+        kwargs = {
+            "session_id": "hello-123",
+            "results_dir": self.tempdir,
+            "cluster": LocalhostCluster(),
```

Comment:
Only issue with this test may be that `LocalhostCluster` is basically the most trivial implementation. Not sure we'll be confident that others would work based on this test.